jupyter_client/multikernelmanager.py

```diff
@@ -78,7 +80,7 @@ def _create_kernel_manager_factory(self):
         def create_kernel_manager(*args, **kwargs):
             km = kernel_manager_ctor(*args, **kwargs)
 
-            if km.transport == 'tcp':
+            if self.cache_ports and km.transport == 'tcp':
```

Comment:
This would then become `if km.cache_ports:` since the default value for `cache_ports` (in the KernelManager) could be `self.transport == 'tcp'`